From 0492d496469823824844ab9bed37cde753e62caa Mon Sep 17 00:00:00 2001
Message-Id: <0492d496469823824844ab9bed37cde753e62caa.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Tue, 20 Jan 2015 17:43:39 +0530
Subject: [PATCH 08/29] MUST_REBASE [VPG]: adf/intel/core: Adding platform id
 check macros

These macros are used to check the platform during runtime.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-5194
Change-Id: I7123dbcd6aa50abbc90bff9143e4384e765c5631
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 .../video/adf/intel/include/core/intel_dc_config.h |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index a6c67e7..203ce74 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -18,6 +18,7 @@
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <video/intel_adf.h>
+#include <drm/i915_adf.h>
 
 #if defined(CONFIG_ADF)
 #include <video/adf.h>
@@ -32,6 +33,21 @@ struct intel_pipe;
 struct intel_dc_memory;
 struct intel_dc_power_ops;
 
+enum gen_id {
+	gen_invalid = 0,
+	/*
+	 * Since adf code supports gen 7 and above
+	 * older gen ids are not kept here
+	 */
+	gen_valleyview = 7,
+	gen_broadwell = 8,
+	gen_cherryview = 8,
+	gen_skylake,
+};
+
+#define IS_CHERRYVIEW() (intel_adf_get_platform_id() == gen_cherryview)
+#define IS_VALLEYVIEW() (intel_adf_get_platform_id() == gen_valleyview)
+
 enum intel_plane_blending {
 	INTEL_PLANE_BLENDING_NONE,
 	INTEL_PLANE_BLENDING_PREMULT,
-- 
1.7.9.5

